lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2233,101 +2234,59 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
-  LogicalResult
-  matchAndRewrite(AtenTransposeIntOp op, ArrayRef<Value> operands,
-                  ConversionPatternRewriter &rewriter) const override {
-    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
-      return failure();
-    AtenTransposeIntOp::Adaptor adaptor(operands);
-
-    int64_t dim0;
-    if (!matchPattern(op.dim0(), m_TorchConstantInt(&dim0)))
-      return rewriter.notifyMatchFailure(op, "dim0 must be constant");
-    int64_t dim1;
-    if (!matchPattern(op.dim1(), m_TorchConstantInt(&dim1)))
-      return rewriter.notifyMatchFailure(op, "dim1 must be constant");
-
-    auto inVector = adaptor.self();
-    auto inType = inVector.getType().cast<RankedTensorType>();
-    auto inputRank = inType.getRank();
-    auto outType = getTypeConverter()
-                       ->convertType(op->getResult(0).getType())
-                       .cast<RankedTensorType>();
-    auto elementType = inType.getElementType();
-
-    dim0 = toPositiveDim(dim0, inputRank);
-    if (!isValidDim(dim0, inputRank))
-      return rewriter.notifyMatchFailure(op, "dim0 out of range");
-    dim1 = toPositiveDim(dim1, inputRank);
-    if (!isValidDim(dim1, inputRank))
-      return rewriter.notifyMatchFailure(op, "dim1 out of range");
-
-    auto loc = op.getLoc();
 
-    SmallVector<Value> outputDims;
-    for (auto i = 0; i < inputRank; i++)
-      outputDims.push_back(getDimOp(rewriter, loc, adaptor.self(), i));
-    std::swap(outputDims[dim0], outputDims[dim1]);
-
-    Value outVector =
-        rewriter.create<linalg::InitTensorOp>(loc, outputDims, elementType);
-    SmallVector<AffineExpr> idExprs;
-    SmallVector<AffineExpr> swapExprs;
-    for (auto i = 0; i < inputRank; i++)
-      idExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
-    for (auto i = 0; i < inputRank; i++) {
-      if (i == dim0)
-        swapExprs.push_back(idExprs[dim1]);
-      else if (i == dim1)
-        swapExprs.push_back(idExprs[dim0]);
-      else
-        swapExprs.push_back(idExprs[i]);
-    }
-
-    SmallVector<AffineMap> indexingMaps = {
-        AffineMap::get(inputRank, 0, idExprs, op.getContext()),
-        AffineMap::get(inputRank, 0, swapExprs, op.getContext())};
-    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
-    auto transpose = rewriter
-                         .create<linalg::GenericOp>(
-                             loc, outVector.getType(), inVector, outVector,
-                             indexingMaps, iteratorTypes,
-                             [](OpBuilder &b, Location loc, ValueRange args) {
-                               b.create<linalg::YieldOp>(loc, args[0]);
-                             })
-                         .getResult(0);
-    rewriter.replaceOpWithNewOp<tensor::CastOp>(op, outType, transpose);
-    return success();
-  }
-};
-} // namespace
-
-namespace {
-class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
+template <typename OpTy>
```

Comment:
I don't think you need to use templates for this. I would instead follow what the `Elementwise` and `Reduction` converters do to maintain a consistent style. In short, the `matchAndRewrite` method can be made to work with all `Op`s.

https://github.com/llvm/torch-mlir/blob/80b15186b10338536325cf62785f572825374e20/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1660